### PR TITLE
Foorm: Add date filter to csv export

### PR DESCRIPTION
**Pull request by molly-moen**
The overall, per-form, foorm csv download currently times out for large datasets. For these, add a date filter to enable smaller csv downloads. This PR also includes some perf improvements to our csv export. However, we still need the date filter as the time to download increases linearly with the number of forms and will eventually time out. This is also a helpful tool for workshop admins, as they can just download the responses they are interested in.

The performance improvements are:
- Avoid some duplicate calculations. For example we were re-parsing the form for every submission, which we do not need to do. Now the form will only be parsed one time. This one change saved about 6 seconds for 1500 submissions.
- Don't try to look for facilitator submissions if we know the form doesn't have facilitator questions (avoids extra queries).

The per-workshop export behavior and UI is unchanged. The overall workshop export modal now looks like this (all fields are required in order for the download button to be enabled):
Screen Shot 2020-10-23 at 4 29 36 PM

There is a help tip for the date range to explain that it is required and advises to shorten the range if the request times out:
Screen Shot 2020-10-23 at 4 29 48 PM

## Links

- [jira](https://codedotorg.atlassian.net/browse/PLC-1035)

## Testing story
Tested locally.

# Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/code-studio/pd/workshop_dashboard/reports/foorm/submissions_download_form.jsx [hunk 1]
```diff
@@ -136,19 +195,77 @@ export default class SubmissionsDownloadForm extends React.Component {
...
+                          <FormGroup>
+                            <ControlLabel>
+                              From
+                              <span data-for="date-tooltip" data-tip>
+                                <FontAwesome
+                                  icon="question-circle-o"
+                                  style={{
+                                    cursor: 'pointer',
+                                    marginLeft: '0.5em',
+                                    marginRight: '0.5em'
```
Comment (breville) [inline apps/src/code-studio/pd/workshop_dashboard/reports/foorm/submissions_download_form.jsx]: Can these go in `styles`?

**Review by breville — APPROVED** [review]
Nice work!

If the timeouts continue to be a problem, does it make sense to consider a slightly different strategy, where we return something immediately while preparing a download asynchronously and then notify the client when it's ready?

File: apps/src/code-studio/pd/workshop_dashboard/reports/foorm/submissions_download_form.jsx (hunk 1 above)
Comment (molly-moen) [reply apps/src/code-studio/pd/workshop_dashboard/reports/foorm/submissions_download_form.jsx]: good call

**Comment by molly-moen** [comment]
@breville I think that the async approach may be a good long-term strategy.

**Pull request merged**